docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md

```diff
@@ -23,48 +10,73 @@ final Properties properties = new Properties();
+### Using the Read-Write Splitting Plugin against non-RDS clusters
 
+If you are using the read-write splitting plugin against a cluster that is not hosted on RDS or Aurora, the plugin will not be able to automatically acquire the cluster topology. Instead, you must supply the topology information in the connection string as a comma-delimited list of multiple instance URLs. If you are using a single writer instance, the first instance in the list must be the writer instance and you must enable the `singleWriterConnectionString` property:
 ```
 properties.setProperty(ConnectionStringHostListProvider.SINGLE_WRITER_CONNECTION_STRING.name, "true");
 String connectionUrl = "jdbc:aws-wrapper:mysql://writer-instance-1.com,reader-instance-1.com,reader-instance-2.com/database-name"
 ```
 
-Additionally, you should avoid using the Aurora host list plugin and the failover plugin in this scenario, as they are designed to specifically operate against Aurora databases.
+Additionally, you should avoid using the Aurora host list plugin and the failover plugin in this scenario, as they are designed to operate against Aurora databases only.
+
+### Limitations
+
+#### General plugin limitations
+
+When a Statement or ResultSet object is created, it is internally bound to the physical database connection established at that moment. There is no standard JDBC functionality to change the underlying connection used by Statement or ResultSet objects. Consequently, even if the read-write plugin switches the internal connection, any Statements/ResultSets created before this event will continue using the old database connection. This  bypasses any desired functionality provided by the read-write plugin. To prevent these scenarios, an exception will be thrown if your code uses any Statements/ResultSets created before a change in internal connection. To solve this problem, please ensure you create new Statement/ResultSet objects after each change in internal connection. See the section on [scenarios that cause a change in internal connection](#scenarios-that-cause-a-change-in-internal-connection) for more info.
+
+#### Multi-statement SQL limitations with reader load balancing
+
+When reader load balancing is enabled and autocommit is off, the read-write splitting plugin will analyze SQL statements executed to determine if `COMMIT` or `ROLLBACK` are executed. In addition to `commit()` and `rollback()`, these statements indicate a transaction boundary. This analysis does not support SQL strings containing multiple statements. If your SQL strings use `COMMIT` or `ROLLBACK` and they contain multiple statements, we recommend that you do not enable reader load balancing as the plugin will not detect the transaction boundary if these keywords are not the first statement.
```

Comment:
I'm wondering if we should remove support for detecting transaction boundaries when `COMMIT` or `ROLLBACK` are executed as SQL. The JBDC spec says [here](https://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html) that `Applications should not invoke SQL commands directly to change the connection's configuration when there is a JDBC method available.` It seems silly and confusing to have to write this limitation paragraph for something that isn't even supported by the JDBC spec. Let me know what you think.

Reply:
I'd keep it. Unfortunately developers don't read JDBC spec. Second line of defence makes sense to me.

Reply:
I see your point. However, to play devil's advocate:
- the mysql driver does not detect executing `COMMIT` or `ROLLBACK` so the expectation that it should work has not been set
- does it make sense to add code to account for scenarios that users shouldn't be following?
- if users complain we can point to the JDBC spec to show they are not following it
- removing this code will simplify the code and the documentation